src/App.js

```diff
@@ -1,21 +1,31 @@
-import { React, useState } from "react";
+
+import { React, useEffect, useState } from "react";
 import ToDoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
+const useSemiPersistentState = () => {
+  const localData = JSON.parse(localStorage.getItem('savedTodoList'))
+  const [todoList, setTodoList] = useState((localData.length ? localData : []))
```

Comment:
There's a bug in how you're setting state here. You are saying if localData.length then return localData, otherwise return an empty array. While you're on the right track, the browser will throw `Uncaught TypeError: Cannot read properties of null (reading 'length')` here when you don't have anything in your localStorage. Arrays and strings are the only two data that implement this property. Thus using it with any other data type, such as null or undefined, will give the error.

You can do your null check when you set your localData variable, and avoid this issue like so:


`const localData = JSON.parse(localStorage.getItem('savedTodoList')) || []
  const [todoList, setTodoList] = useState((localData))`